cosmos/config.py

```diff
@@ -267,35 +299,40 @@ def ensure_profile(
             yield Path(self.profiles_yml_filepath), {}
 
         elif self.profile_mapping:
-            profile_contents = self.profile_mapping.get_profile_file_contents(
-                profile_name=self.profile_name, target_name=self.target_name, use_mock_values=use_mock_values
-            )
-
             if use_mock_values:
                 env_vars = {}
             else:
                 env_vars = self.profile_mapping.env_vars
 
-            if desired_profile_path:
-                logger.info(
-                    "Writing profile to %s with the following contents:\n%s",
-                    desired_profile_path,
-                    profile_contents,
-                )
-                # write profile_contents to desired_profile_path using yaml library
-                desired_profile_path.write_text(profile_contents)
-                yield desired_profile_path, env_vars
+            if enable_profile_cache:
```

Comment:
This should be an end before the global setting `enable_cache` and the specific profile cache

Reply:
do you mean we should also check if `enable_cache` is also enable or something more?